File: sfn/nested/SfnMerge.cs
namespace sfn_ut.sfn.nested;

public abstract class SfnMerge<PARENT,CHILD> : SfnStep {
  public abstract PARENT Execute(SfnMergeInput<PARENT,CHILD> input);

  public SfnMerge() : base(StepType.Merge) {
    // Do nothing
  }
}

File: java-sfn-unit-test/src/main/java/com/sfnut/sfn/nested/SfnMerge.java
package com.sfnut.sfn.nested;

import com.sfnut.sfn.SfnStep;
import com.sfnut.sfn.StepType;

public abstract class SfnMerge<PARENT, CHILD> extends SfnStep {
    
    public SfnMerge() {
        super(StepType.OUTPUT_ADAPTER);
    }

    public abstract PARENT execute(SfnMergeInput<PARENT, CHILD> input);
}
